src/addons/manila_csi.rs

```diff
@@ -0,0 +1,326 @@
+use crate::{
+    addons::{ClusterAddon, ClusterAddonValues, ClusterAddonValuesError, ImageDetails},
+    magnum::{self, ClusterError},
+};
+use docker_image::DockerImage;
+use include_dir::include_dir;
+use k8s_openapi::api::core::v1::Toleration;
+use maplit::btreemap;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+pub struct CSIValues {
+    csimanila: CSIComponent,
+    nodeplugin: CSINodePlugin,
+    controllerplugin: CSIControllerPlugin,
+}
+
+#[derive(Debug, Deserialize, PartialEq, Serialize)]
+struct CSIComponent {
+    image: ImageDetails,
+}
```

Comment:
Is this the same CSIComponent used in Cinder?  If so, let's create a `csi.rs` module and extract it to in order to refactor and not duplicate stuff :)